src/devices/acurite.c

```diff
@@ -490,55 +435,80 @@ static int acurite_txr_callback(bitbuffer_t *bitbuf) {
 
 	// The 5-n-1 weather sensor messages are 8 bytes.
 	if (browlen == ACURITE_5N1_BITLEN / 8) {
+        if (debug_output) {
+            fprintf(stderr, "Acurite 5n1 raw msg: %02X %02X %02X %02X %02X %02X %02X %02X\n",
+                bb[0], bb[1], bb[2], bb[3], bb[4], bb[5], bb[6], bb[7]);
+        }
 	    channel = acurite_getChannel(bb[0]);
+        sprintf(channel_str, "%c", channel);        
 	    sensor_id = acurite_5n1_getSensorId(bb[0],bb[1]);
 	    repeat_no = acurite_5n1_getMessageCaught(bb[0]);
 	    message_type = bb[2] & 0x3f;
-
+        battery_low = (bb[2] & 0x40) >> 6;
 
 	    if (message_type == 0x31) {
-		// Wind speed, wind direction, and rain fall
-	        wind_speed = acurite_getWindSpeed(bb[3], bb[4]);
-		wind_speedmph = kmph2mph(wind_speed);
-		wind_dird = acurite_5n1_winddirections[bb[4] & 0x0f];
-		wind_dirstr = acurite_5n1_winddirection_str[bb[4] & 0x0f];
-		raincounter = acurite_getRainfallCounter(bb[5], bb[6]);
-		if (acurite_5n1t_raincounter > 0) {
-		    // track rainfall difference after first run
-		    // FIXME when converting to structured output, just output
-		    // the reading, let consumer track state/wrap around, etc. 
-		    rainfall = ( raincounter - acurite_5n1t_raincounter ) * 0.01;
-		    if (raincounter < acurite_5n1t_raincounter) {
-			printf("%s Acurite 5n1 sensor 0x%04X Ch %c, rain counter reset or wrapped around (old %d, new %d)\n",
-			       time_str, sensor_id, channel, acurite_5n1t_raincounter, raincounter);
-			acurite_5n1t_raincounter = raincounter;
-		    }
-		} else {
-		    // capture starting counter
-		    acurite_5n1t_raincounter = raincounter;
-		    printf("%s Acurite 5n1 sensor 0x%04X Ch %c, Total rain fall since last reset: %0.2f\n",
-			   time_str, sensor_id, channel, raincounter * 0.01);
-		}
-
-		printf("%s Acurite 5n1 sensor 0x%04X Ch %c, Msg %02x, Wind %d kmph / %0.1f mph %0.1f° %s (%d), rain gauge %0.2f in.\n",
-		       time_str, sensor_id, channel, message_type,
-		       wind_speed, wind_speedmph,
-		       wind_dird, wind_dirstr, bb[4] & 0x0f, rainfall);
+            // Wind speed, wind direction, and rain fall
+            wind_speed = acurite_getWindSpeed_kph(bb[3], bb[4]);
+            wind_speedmph = kmph2mph(wind_speed);
+            wind_dird = acurite_5n1_winddirections[bb[4] & 0x0f];
+            wind_dirstr = acurite_5n1_winddirection_str[bb[4] & 0x0f];
+            raincounter = acurite_getRainfallCounter(bb[5], bb[6]);
+            if (acurite_5n1t_raincounter > 0) {
+                // track rainfall difference after first run
+                // FIXME when converting to structured output, just output
+                // the reading, let consumer track state/wrap around, etc. 
+                rainfall = ( raincounter - acurite_5n1t_raincounter ) * 0.01;
```

Comment:
What about outputting both the accumulated value (as recorded since the start of rtl_433) and the raw value (i.e. number of bucket tips, e.g. for external processing) - this should be easy enough.

Note that I have already discovered a big bug in the rain accumulation code - if there is more than one 5n1 (vn1tx) rtl_433 will get confused because it's not treating these as separate instances and constantly flip-flopping between them, example:
```
acurite_txr Parity: 11000000
{"time" : "2017-01-07 12:18:00", "model" : "Acurite 5n1 sensor", "sensor_id" : 3585, "channel" : "B", "battery" : "LOW", "message_type" : 56, "wind_speed" : 0.000, "temperature_C" : 18.778, "humidity" : 29}
acurite_txr Parity: 00000000
2017-01-07 12:18:15 Acurite 5n1 sensor 0x0688 Ch A, Total rain fall since last reset: 103.15
{"time" : "2017-01-07 12:18:15", "model" : "Acurite 5n1 sensor", "sensor_id" : 1672, "channel" : "A", "battery" : "LOW", "message_type" : 49, "wind_speed" : 5.139, "wind_dir_deg" : 22.500, "wind_dir" : "NNE", "rainfall" : 0.000}
acurite_txr Parity: 10000000
{"time" : "2017-01-07 12:18:15", "model" : "Acurite 5n1 sensor", "sensor_id" : 1672, "channel" : "A", "battery" : "LOW", "message_type" : 49, "wind_speed" : 5.139, "wind_dir_deg" : 22.500, "wind_dir" : "NNE", "rainfall" : 0.000}
acurite_txr Parity: 10000001
{"time" : "2017-01-07 12:18:15", "model" : "Acurite 5n1 sensor", "sensor_id" : 1672, "channel" : "A", "battery" : "LOW", "message_type" : 49, "wind_speed" : 5.139, "wind_dir_deg" : 22.500, "wind_dir" : "NNE", "rainfall" : 0.000}
acurite_txr Parity: 01000001
2017-01-07 12:18:19 Acurite 5n1 sensor 0x0E01 Ch B, rain counter reset or wrapped around (old 10315, new 0)
{"time" : "2017-01-07 12:18:19", "model" : "Acurite 5n1 sensor", "sensor_id" : 3585, "channel" : "B", "battery" : "LOW", "message_type" : 49, "wind_speed" : 0.000, "wind_dir_deg" : 0.000, "wind_dir" : "N", "rainfall" : -103.150}
acurite_txr Parity: 11000000
2017-01-07 12:18:19 Acurite 5n1 sensor 0x0E01 Ch B, Total rain fall since last reset: 0.00
{"time" : "2017-01-07 12:18:19", "model" : "Acurite 5n1 sensor", "sensor_id" : 3585, "channel" : "B", "battery" : "LOW", "message_type" : 49, "wind_speed" : 0.000, "wind_dir_deg" : 0.000, "wind_dir" : "N", "rainfall" : -103.150}
acurite_txr Parity: 11000000
2017-01-07 12:18:19 Acurite 5n1 sensor 0x0E01 Ch B, Total rain fall since last reset: 0.00
{"time" : "2017-01-07 12:18:19", "model" : "Acurite 5n1 sensor", "sensor_id" : 3585, "channel" : "B", "battery" : "LOW", "message_type" : 49, "wind_speed" : 0.000, "wind_dir_deg" : 0.000, "wind_dir" : "N", "rainfall" : -103.150}
```